pandas/io/parsers.py

```diff
@@ -3264,6 +3266,9 @@ def _make_date_converter(
 ):
     def converter(*date_cols):
         if date_parser is None:
+            date_cols = tuple(
+                x if isinstance(x, np.ndarray) else x.to_numpy() for x in date_cols
+            )
```

Comment:
this is better off done *inside* concat_date_cols, but what is the incoming data here in the example?

Reply:
It's tuple with `StringArray` with dates from `'b'`.